comp/core/telemetry/component.go

```diff
@@ -0,0 +1,73 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+// Package telemetry implements a component for all agent telemetry.
+package telemetry
+
+import (
+	"net/http"
+
+	"go.opentelemetry.io/otel/metric"
+	"go.uber.org/fx"
+
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+)
+
+// team: agent-shared-components
+
+// Component is the component type.
+type Component interface {
+	// Handler returns an http handler to expose the internal metrics
+	Handler() http.Handler
+	// Reset resets all tracked telemetry
+	Reset()
+	// Meter returns a new OTEL meter
+	Meter(name string, opts ...metric.MeterOption) metric.Meter
+
+	// NewCounter creates a Counter with default options for telemetry purpose.
+	NewCounter(subsystem, name string, tags []string, help string) Counter
+	// NewCounterWithOpts creates a Counter with the given options for telemetry purpose.
+	NewCounterWithOpts(subsystem, name string, tags []string, help string, opts Options) Counter
```

Comment:
That's a nice API design. I see 3 possibilities: 
1. we leave it as is (best compatibility with existing code)
2. we can use the functional options (this is nice and clean)
3. we can adopt the `KeyValue` style attributes (that OTEL uses) so we are more compatible with the newer API. 

I chose 1 in this case because the compatibility is nice and we are introducing a new way with OTEL anyway. WDYT?